### PR TITLE
Leo dev

### PR DESCRIPTION
**Pull request by LeoPCaucheteux**

### CODE REVIEW

File: src/generic.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        path (string): path where the model is saved
+        model: model to save
+        name (string): name of the model you want to loan
+    Returns:
+        
+    """
+    model_path=os.path.join(path.resolve().as_posix(), name)
+    pickle.dump(model, open(model_path, 'wb'))
+    print("Model has been saved in the following location",model_path)
+    return()
```
Comment (griffreichert) [inline src/generic.py]: does this need to return?

Comment (LeoPCaucheteux) [reply src/generic.py]: No, the pickle.dump is doing the action.
We could return the print message though

File: src/DT_Data_Processing.py
Comment (griffreichert) [inline src/DT_Data_Processing.py:1]: can we name this file 'decision_tree.py', best to use all lowercase for these to make imports easy

File: src/generic.py
Comment (griffreichert) [inline src/generic.py:1]: can we name this something more clear like 'model_utils.py', i'm also fine if the code goes in data setup

File: src/visualisation.py
Comment (griffreichert) [inline src/visualisation.py:1]: should build_hierarchical_dataframe() go in data setup?

File: src/DT_Data_Processing.py
Comment (griffreichert) [reply src/DT_Data_Processing.py:1]: Also we could use the data_setup.py date feature methods that exist already for consistency

**Review by griffreichert — CHANGES_REQUESTED** [review]
thanks for adding this leo, added a few naming and organisation suggestions

**Review by griffreichert — APPROVED** [review]

File: src/visualisation.py
Comment (LeoPCaucheteux) [reply src/visualisation.py:1]: We can move it afterward but yeah it would make sense to have it there

**Pull request merged**